Menu optimizations

leptun opened the pull request:
Various menu optimizations

leptun committed:
Remove duplicate separator string

Flash: -14B
SRAM: 0B

leptun committed:
Correctly utilize the menu stack in the Mode select message

Flash: -34B
SRAM: 0B

leptun committed:
Dead code cleanup

gudnimg commented on Firmware/ultralcd.cpp:
```diff
@@ -3456,8 +3441,7 @@ static void crash_mode_switch()
     }
-	if (printJobOngoing() || (lcd_commands_type == LcdCommands::Layer1Cal)) menu_goto(lcd_tune_menu, 9, true, true);
-	else menu_goto(lcd_settings_menu, 9, true, true);
+    menu_back();
```
The call to `menu_back()` should be removed. There is no need for a menu change here :)

gudnimg committed:
Fix an issue with crash_mode_switch toggling

Proposed changes

gudnimg commented on Firmware/ultralcd.cpp:
```diff
@@ -3423,10 +3411,7 @@ static void lcd_silent_mode_set() {
   // Wait until the planner queue is drained and the stepper routine achieves
   // an idle state.
   st_synchronize();
-  if (tmc2130_wait_standstill_xy(1000)) {}
```
Proposal for change:

Guard the `lcd_display_message_fullscreen_P` call by `blocks_queued()`. If there are no planned moves, there is no need to render this message, it creates a weird LCD draw noticable by the user.

This does add about 10B of flash but I think its worth it for slightly cleaner UI. I'm not sure if `tmc2130_wait_standstill_xy` should be guarded as well, you're more familiar with it than me.

![image](https://user-images.githubusercontent.com/8218499/233787897-d0ae5e0e-7523-49e7-b8ae-8e6bdcb67a79.png)

gudnimg committed:
Don't render message when printer is idle

Guard the lcd_display_message_fullscreen_P call by blocks_queued(). If there are no planned moves, there is no need to render this message, it creates a weird LCD draw noticable by the user.

Change in memory:
Flash: +10 bytes
SRAM: 0 bytes

leptun committed:
Remove AUTOTEMP menu items

They are commented out and broken anyway

leptun committed:
Remove stupid comments from older bad code

leptun committed:
Deduplicate MMU settings

Flash: -58B
SRAM: 0B

leptun committed:
Deduplicate temperature settings

Flash: -52B
SRAM: 0B
